src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -2788,6 +2788,14 @@ internal PropertyCacheEntry(PropertyInfo property)
                 if (propertySetter != null && (propertySetter.IsPublic || propertySetter.IsFamily))
                 {
                     this.isStatic = propertySetter.IsStatic;
+                    foreach (var requiredCustomModifier in propertySetter.ReturnParameter.GetRequiredCustomModifiers())
+                    {
+                        if (!requiredCustomModifier.IsNested && requiredCustomModifier.Name == "IsExternalInit" && requiredCustomModifier.Namespace == "System.Runtime.CompilerServices")
```

Comment:
Based on https://github.com/dotnet/roslyn/issues/50696#issuecomment-782188548  I guess we shouldn't check `System.Runtime.CompilerServices`.

Reply:
> Based on dotnet/roslyn#50696 (comment) I guess we shouldn't check `System.Runtime.CompilerServices`.

They are referring to assembly identity there. So you don't want to check `== typeof(IsExternalInit)`, but the compiler will still emit the full type name outside of corelib

Reply:
Does it make sense to test if it's `sealed`?

Reply:
> Does it make sense to test if it's sealed?

Just noticed there are multiple "backwards-compatibility" libraries on nuget already that **don't** mark the class `sealed`, including [this one referenced by roslyn](https://github.com/manuelroemer/IsExternalInit/blob/master/src/IsExternalInit/IsExternalInit.cs), so I guess that's a resounding NO :)